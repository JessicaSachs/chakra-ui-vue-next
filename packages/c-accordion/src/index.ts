import { h, defineComponent, PropType } from 'vue'
import { chakra, DOMElements } from '@chakra-ui/vue-system'

const CAccordion = defineComponent({
  props: {
    as: {
      type: [String] as PropType<DOMElements>,
      default: 'div',
    },
  },
  setup(props, { slots, attrs }) {
    return () => h(chakra(props.as), { ...attrs }, slots)
  },
})

export default CAccordion
