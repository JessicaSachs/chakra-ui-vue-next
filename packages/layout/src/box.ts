import { chakra, DOMElements, SystemStyleObject } from '@chakra-ui/vue-system'
import { defineComponent, h, PropType } from '@vue/runtime-core'

// export interface BoxProps extends HTMLChakraProps<"div"> {}
interface BoxProps {
  width: object | string | number
}

// export const CBox = chakra('div') // as prop doesnt work this way

/**
 * Box is the most abstract component on top of which other chakra
 * components are built. It renders a `div` element by default.
 *
 * @see Docs https://vue.chakra-ui.com/docs/layout/box
 */
export const CBox = defineComponent({
  props: {
    as: {
      type: [String, Object] as PropType<DOMElements>,
      default: 'div',
    },
  },
  setup(props, { slots, attrs }) {
    return () => {
      return h(
        chakra(props.as, {
          label: attrs.label ? (attrs.label as string) : 'box', // for CSquare
        }),
        {
          ...attrs,
        },
        slots
      )
    }
  },
})

interface SquareProps {
  /**
   * The size (width and height) of the square
   */
  size?: BoxProps['width']
  /**
   * If `true`, the content will be centered in the square
   */
  centerContent?: boolean
}
export const CSquare = defineComponent({
  props: {
    size: [Object, String, Number] as PropType<SquareProps['size']>,
    centerContent: {
      type: [Boolean] as PropType<SquareProps['centerContent']>,
      default: true,
    },
  },
  setup(props, { slots, attrs }) {
    const styles: SystemStyleObject = props.centerContent
      ? { display: 'flex', alignItems: 'center', justifyContent: 'center' }
      : {}
    return () => {
      return h(
        chakra(CBox, {
          label: attrs.label ? (attrs.label as string) : 'square', // for CCircle
          boxSize: props.size,
          __css: {
            ...styles,
            flexShrink: 0,
            flexGrow: 0,
          },
          ...attrs,
        }),
        {},
        slots
      )
    }
  },
})

export const CCircle = defineComponent({
  props: {
    size: [Object, String, Number] as PropType<SquareProps['size']>,
    centerContent: {
      type: [Boolean] as PropType<SquareProps['centerContent']>,
      default: true,
    },
  },
  setup(props, { slots, attrs }) {
    return () => {
      return h(
        CSquare,
        {
          label: 'circle',
          borderRadius: '9999px',
          size: props.size,
          ...attrs,
        },
        slots
      )
    }
  },
})
