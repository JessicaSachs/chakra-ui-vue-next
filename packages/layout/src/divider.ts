import { h, defineComponent, PropType, computed } from 'vue'
import {
  chakra,
  ThemingProps,
  useStyleConfig,
  HTMLChakraProps,
  extractStyleAttrs,
} from '@chakra-ui/vue-system'
import { filterUndefined } from '@chakra-ui/utils'
import { vueThemingProps } from '@chakra-ui/vue-utils'

export interface DividerProps
  extends HTMLChakraProps<'div'>,
    ThemingProps<'Container'> {
  orientation?: 'horizontal' | 'vertical'
}

/**
 * Layout component used to visually separate content in a list or group.
 * It display a thin horizontal or vertical line, and renders a `hr` tag.
 *
 * @see Docs https://vue.chakra-ui.com/docs/data-display/divider
 */
export const CDivider = defineComponent({
  props: {
    orientation: {
      type: [String] as PropType<DividerProps['orientation']>,
      default: 'horizontal',
    },
    ...vueThemingProps,
  },
  setup(props, { slots, attrs }) {
    return () => {
      const themingProps = computed<ThemingProps>(() =>
        filterUndefined({
          colorScheme: props.colorScheme,
          styleConfig: props.styleConfig,
          orientation: props.orientation,
        })
      )
      // to support border attrs like borderBottomWidth etc
      const styleAttrs = extractStyleAttrs(attrs).styles

      const allProps = {
        ...themingProps.value,
        ...styleAttrs,
      }

      const styles = useStyleConfig('Divider', allProps as DividerProps)
      const {
        borderLeftWidth,
        borderBottomWidth,
        borderTopWidth,
        borderRightWidth,
        borderWidth,
        borderStyle,
        borderColor,
        ...stylesRest
      } = styles.value

      const dividerStyles = {
        vertical: {
          borderLeftWidth:
            borderLeftWidth || borderRightWidth || borderWidth || '1px',
          height: '100%',
        },
        horizontal: {
          borderBottomWidth:
            borderBottomWidth || borderTopWidth || borderWidth || '1px',
          width: '100%',
        },
      }

      return h(
        chakra('hr', { label: 'divider' }),
        {
          'aria-orientation': props.orientation,
          ...styles.value,
          __css: {
            ...stylesRest,
            border: '0',

            borderColor,
            borderStyle,
            ...dividerStyles[props.orientation!],
          },
          ...attrs,
        },
        slots
      )
    }
  },
})
