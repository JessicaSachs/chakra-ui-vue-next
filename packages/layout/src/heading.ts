import { vueThemingProps } from '@chakra-ui/vue-utils'
import {
  chakra,
  DOMElements,
  ThemingProps,
  useStyleConfig,
  HTMLChakraProps,
  extractStyleAttrs,
} from '@chakra-ui/vue-system'
import { computed, defineComponent, h, PropType } from '@vue/runtime-core'
import { filterUndefined } from '@chakra-ui/utils'

export interface HeadingProps
  extends HTMLChakraProps<'h2'>,
    ThemingProps<'Heading'> {}

export const CHeading = defineComponent({
  props: {
    as: {
      type: [String, Object] as PropType<DOMElements>,
      default: 'h2',
    },
    ...vueThemingProps,
  },
  setup(props, { slots, attrs }) {
    const themingProps = computed<ThemingProps>(() =>
      filterUndefined({
        colorScheme: props.colorScheme,
        variant: props.variant,
        size: props.size,
        styleConfig: props.styleConfig,
      })
    )
    const styles = useStyleConfig('Heading', themingProps.value)
    // convert attrs like `is-truncated` to `isTruncated`
    const styleAttrs = extractStyleAttrs(attrs).styles as any

    return () => {
      return h(
        chakra(props.as, {
          label: 'heading',
          ...styleAttrs,
          __css: styles.value,
        }),
        {},
        slots
      )
    }
  },
})
