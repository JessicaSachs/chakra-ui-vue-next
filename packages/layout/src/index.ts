export * from './link'
export * from './badge'
export * from './stack'
