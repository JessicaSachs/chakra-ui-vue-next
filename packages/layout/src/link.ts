import { vueThemingProps } from './utils'
import { h, defineComponent, PropType, computed } from 'vue'
import {
  chakra,
  DOMElements,
  ThemingProps,
  useStyleConfig,
} from '@chakra-ui/vue-system'
import { filterUndefined } from '@chakra-ui/utils'

/**
 * Links are accessible elements used primarily for navigation.
 *
 * It integrates well with other routing libraries like
 * React Router, Reach Router and Next.js Link.
 *
 * @example
 *
 * ```jsx
 * <Link as={ReactRouterLink} to="/home">Home</Link>
 * ```
 *
 * @see Docs https://chakra-ui.com/docs/layout/link
 */
export const CLink = defineComponent({
  props: {
    as: {
      type: [Object, String] as PropType<DOMElements>,
      default: 'a',
    },
    isExternal: Boolean,
    ...vueThemingProps,
  },
  setup(props, { slots, attrs }) {
    return () => {
      const themingProps = computed<ThemingProps>(() =>
        filterUndefined({
          colorScheme: props.colorScheme,
          variant: props.variant,
          size: props.size,
          styleConfig: props.styleConfig,
        })
      )
      const styles = useStyleConfig('Link', themingProps.value)

      return h(
        chakra(props.as),
        {
          target: props.isExternal ? '_blank' : undefined,
          rel: props.isExternal ? 'noopener noreferrer' : undefined,
          __css: styles.value,
          ...attrs,
        },
        slots
      )
    }
  },
})
