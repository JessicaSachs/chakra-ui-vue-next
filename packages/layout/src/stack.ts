import { SystemProps } from '@chakra-ui/styled-system'
import {
  h,
  defineComponent,
  PropType,
  Component,
  computed,
  Fragment,
  createVNode,
} from 'vue'
import { chakra, DOMElements } from '@chakra-ui/vue-system'
import {
  getDividerStyles,
  getStackStyles,
  selector,
  StackDirection,
} from './stack.utils'
import { getValidChildren } from '@chakra-ui/vue-utils'

interface StackProps {
  /**
   * Shorthand for `alignItems` style prop
   * @type SystemStyleObject["alignItems"]
   */
  align?: SystemProps['alignItems']
  /**
   * Shorthand for `justifyContent` style prop
   * @type SystemStyleObject["justifyContent"]
   */
  justify?: SystemProps['justifyContent']
  /**
   * Shorthand for `flexWrap` style prop
   * @type SystemStyleObject["flexWrap"]
   */
  wrap?: SystemProps['flexWrap']
  /**
   * The space between each stack item
   * @type SystemStyleObject["margin"]
   */
  spacing?: SystemProps['margin']
  /**
   * The direction to stack the items.
   */
  direction?: StackDirection
  /**
   * If `true`, each stack item will show a divider
   * @type React.ReactElement
   */
  divider?: Component
  /**
   * If `true`, the children will be wrapped in a `Box` with
   * `display: inline-block`, and the `Box` will take the spacing props
   */
  shouldWrapChildren?: boolean
  /**
   * If `true` the items will be stacked horizontally.
   */
  isInline?: boolean
}

export const CStackDivider = defineComponent({
  setup(_, { attrs }) {
    console.log(attrs.__css)
    return () => {
      return h(
        chakra('div', {
          label: 'stack__divider',
          borderWidth: 0,
          alignSelf: 'stretch',
          borderColor: 'inherit',
          width: 'auto',
          height: 'auto',
          __css: attrs.__css as any,
        })
      )
    }
  },
})

export const CStackItem = defineComponent({
  setup(_, { attrs }) {
    return () => {
      return h(
        chakra('div', {
          label: 'stack__item',
          ...attrs,
          display: 'inline-block',
          flex: '0 0 auto',
          minWidth: 0,
          __css: attrs.__css as any,
        })
      )
    }
  },
})

const stackProps = {
  as: {
    type: [Object, String] as PropType<DOMElements>,
    default: 'div',
  },
  align: [Object, String, Array] as PropType<StackProps['align']>,
  justify: [Object, String, Array] as PropType<StackProps['justify']>,
  wrap: [Object, String, Array] as PropType<StackProps['wrap']>,
  spacing: {
    type: [Object, String, Array] as PropType<StackProps['spacing']>,
    default: '0.5rem',
  },
  direction: {
    type: [Object, String, Array] as PropType<StackProps['direction']>,
    default: 'row',
  },
  // todo: divider
  divider: [Object, Boolean] as PropType<StackProps['divider']>,
  shouldWrapChildren: [Boolean] as PropType<StackProps['shouldWrapChildren']>,
  isInline: [Boolean] as PropType<StackProps['isInline']>,
}

export const CStack = defineComponent({
  name: 'CStack',
  props: stackProps,
  setup(props, { slots, attrs }) {
    const direction = computed(() =>
      props.isInline ? 'row' : props.direction ?? 'column'
    )

    const styles = computed(() =>
      getStackStyles({ direction: direction.value, spacing: props.spacing })
    )

    const dividerStyle = computed(() =>
      getDividerStyles({ spacing: props.spacing, direction: direction.value })
    )

    const hasDivider = computed(() => !!props.divider)

    const shouldUseChildren = computed(
      () => !props.shouldWrapChildren && !hasDivider.value
    )

    return () => {
      const validChildren = getValidChildren(slots)
      const clones = shouldUseChildren.value
        ? validChildren
        : validChildren.map((child, index) => {
            const isLast = index + 1 === validChildren.length
            const wrappedChild = createVNode(CStackItem, { key: index }, child)
            const _child = props.shouldWrapChildren ? wrappedChild : child

            if (!hasDivider.value) return _child

            // todo: temporary divider
            const clonedDivider = createVNode(CStackDivider, {
              borderColor: 'blue.200',
              __css: dividerStyle.value,
            })

            const _divider = isLast ? null : clonedDivider

            return createVNode(Fragment, { key: index }, [_child, _divider])
          })

      return h(
        chakra('div', {
          label: attrs.label ? (attrs.label as string) : 'stack', // CHStack and CVStack
          display: 'flex',
          alignItems: props.align,
          justifyContent: props.justify,
          flexDirection: styles.value.flexDirection,
          flexWrap: props.wrap,
          __css: hasDivider.value ? {} : { [selector]: styles.value[selector] },
        }),
        () => clones
      )
    }
  },
})

// will change this with HTMLChakraProps
type SystemAndStackProps = StackProps & SystemProps

/**
 * A view that arranges its children in a horizontal line.
 */
export const CHStack = defineComponent({
  name: 'CHStack',
  props: stackProps,
  setup(props, { attrs, slots }) {
    return () => {
      return h(
        CStack,
        {
          label: 'stack-horizontal',
          align: 'center',
          ...props,
          ...attrs,
          direction: 'row',
        } as SystemAndStackProps,
        slots
      )
    }
  },
})

/**
 * A view that arranges its children in a vertical line.
 */
export const CVStack = defineComponent({
  name: 'CVStack',
  props: stackProps,
  setup(props, { attrs, slots }) {
    return () => {
      return h(
        CStack,
        {
          label: 'stack-vertical',
          align: 'center',
          ...props,
          ...attrs,
          direction: 'column',
        } as SystemAndStackProps,
        slots
      )
    }
  },
})
