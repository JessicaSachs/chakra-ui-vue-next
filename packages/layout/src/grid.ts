import { BoxProps } from './box'
import { h, defineComponent, PropType } from 'vue'
import {
  chakra,
  HTMLChakraProps,
  SystemProps,
  ResponsiveValue,
} from '@chakra-ui/vue-system'
import { filterUndefined, mapResponsive } from '@chakra-ui/utils'

export interface GridProps extends HTMLChakraProps<'div'>, GridOptions {}

export interface GridOptions {
  /**
   * Short hand prop for `gridTemplateColumns`
   * @type SystemProps["gridTemplateColumns"]
   */
  templateColumns?: SystemProps['gridTemplateColumns']
  /**
   * Short hand prop for `gridGap`
   * @type SystemProps["gridGap"]
   */
  gap?: SystemProps['gridGap']
  /**
   * Short hand prop for `gridRowGap`
   * @type SystemProps["gridRowGap"]
   */
  rowGap?: SystemProps['gridRowGap']
  /**
   * Short hand prop for `gridColumnGap`
   * @type SystemProps["gridColumnGap"]
   */
  columnGap?: SystemProps['gridColumnGap']
  /**
   * Short hand prop for `gridAutoFlow`
   * @type SystemProps["gridAutoFlow"]
   */
  autoFlow?: SystemProps['gridAutoFlow']
  /**
   * Short hand prop for `gridAutoRows`
   * @type SystemProps["gridAutoRows"]
   */
  autoRows?: SystemProps['gridAutoRows']
  /**
   * Short hand prop for `gridAutoColumns`
   * @type SystemProps["gridAutoColumns"]
   */
  autoColumns?: SystemProps['gridAutoColumns']
  /**
   * Short hand prop for `gridTemplateRows`
   * @type SystemProps["gridTemplateRows"]
   */
  templateRows?: SystemProps['gridTemplateRows']
  /**
   * Short hand prop for `gridTemplateAreas`
   * @type SystemProps["gridTemplateAreas"]
   */
  templateAreas?: SystemProps['gridTemplateAreas']
  /**
   * Short hand prop for `gridArea`
   * @type SystemProps["gridArea"]
   */
  area?: SystemProps['gridArea']
  /**
   * Short hand prop for `gridColumn`
   * @type SystemProps["gridColumn"]
   */
  column?: SystemProps['gridColumn']
  /**
   * Short hand prop for `gridRow`
   * @type SystemProps["gridRow"]
   */
  row?: SystemProps['gridRow']
}

export interface GridItemProps extends BoxProps {
  /**
   * The number of columns the grid item should `span`.
   * @type ResponsiveValue<number | "auto">
   */
  colSpan?: ResponsiveValue<number | 'auto'>
  /**
   * The column number the grid item should start.
   * @type ResponsiveValue<number | "auto">
   */
  colStart?: ResponsiveValue<number | 'auto'>
  /**
   * @type ResponsiveValue<number | "auto">
   */
  colEnd?: ResponsiveValue<number | 'auto'>
  /**
   * @type ResponsiveValue<number | "auto">
   */
  rowStart?: ResponsiveValue<number | 'auto'>
  /**
   * @type ResponsiveValue<number | "auto">
   */
  rowEnd?: ResponsiveValue<number | 'auto'>
  /**
   * @type ResponsiveValue<number | "auto">
   */
  rowSpan?: ResponsiveValue<number | 'auto'>
}

/**
 * Vue component used to create grid layouts.
 *
 * It renders a `div` with `display: grid` and
 * comes with helpful style shorthand.
 *
 * @see Docs https://vue.chakra-ui.com/docs/layout/grid
 */
export const CGrid = defineComponent({
  props: {
    templateColumns: [Object, String, Array] as PropType<
      GridProps['gridTemplateColumns']
    >,
    gap: [Object, String, Array] as PropType<GridProps['gridGap']>,
    rowGap: [Object, String, Array] as PropType<GridProps['gridRowGap']>,
    columnGap: [Object, String, Array] as PropType<GridProps['gridColumnGap']>,
    autoFlow: [Object, String, Array] as PropType<GridProps['gridAutoFlow']>,
    autoRows: [Object, String, Array] as PropType<GridProps['gridAutoRows']>,
    autoColumns: [Object, String, Array] as PropType<
      GridProps['gridAutoColumns']
    >,
    templateRows: [Object, String, Array] as PropType<
      GridProps['gridTemplateRows']
    >,
    templateAreas: [Object, String, Array] as PropType<
      GridProps['gridTemplateAreas']
    >,
    area: [Object, String, Array] as PropType<GridProps['gridArea']>,
    column: [Object, String, Array] as PropType<GridProps['gridColumn']>,
    row: [Object, String, Array] as PropType<GridProps['gridRow']>,
  },
  setup(props, { slots, attrs }) {
    return () => {
      const styles = filterUndefined({
        display: 'grid',
        gridArea: props.area,
        gridTemplateAreas: props.templateAreas,
        gridGap: props.gap,
        gridRowGap: props.rowGap,
        gridColumnGap: props.columnGap,
        gridAutoColumns: props.autoColumns,
        gridColumn: props.column,
        gridRow: props.row,
        gridAutoFlow: props.autoFlow,
        gridAutoRows: props.autoRows,
        gridTemplateRows: props.templateRows,
        gridTemplateColumns: props.templateColumns,
      })

      return h(
        chakra('div', { label: 'grid' }),
        {
          __css: styles,
          ...attrs,
        },
        slots
      )
    }
  },
})

function spanFn(span?: ResponsiveValue<number | 'auto'>) {
  return mapResponsive(span, (value) =>
    value === 'auto' ? 'auto' : `span ${value}/span ${value}`
  )
}

export const CGridItem = defineComponent({
  props: {
    colSpan: [Object, String, Array] as PropType<GridItemProps['colSpan']>,
    colStart: [Object, String, Array] as PropType<GridItemProps['colStart']>,
    colEnd: [Object, String, Array] as PropType<GridItemProps['colEnd']>,
    rowStart: [Object, String, Array] as PropType<GridItemProps['rowStart']>,
    rowEnd: [Object, String, Array] as PropType<GridItemProps['rowEnd']>,
    rowSpan: [Object, String, Array] as PropType<GridItemProps['rowSpan']>,
  },
  setup(props, { slots, attrs }) {
    return () => {
      const styles = filterUndefined({
        gridColumn: spanFn(props.colSpan),
        gridRow: spanFn(props.rowSpan),
        gridColumnStart: props.colStart,
        gridColumnEnd: props.colEnd,
        gridRowStart: props.rowStart,
        gridRowEnd: props.rowEnd,
      })

      return h(
        chakra('div', { label: 'grid__item' }),
        {
          __css: styles,
          ...attrs,
        },
        slots
      )
    }
  },
})
