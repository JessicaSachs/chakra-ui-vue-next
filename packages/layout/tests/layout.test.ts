import { CAspectRatio, CLink, CBadge, CStack, CVStack, CHStack } from '../src'
import { render, testA11y, TestRenderProps } from '../../test-utils/src'

describe('<CLink />', () => {
  const renderComponent = () =>
    render({
      components: { CLink },
      template: `<CLink>CLink</CLink>
      <CLink is-external>CLink external</CLink>
      `,
    })

  it('should render properly', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('a11y test', async () => {
    await testA11y(renderComponent())
  })
})

describe('<CBadge />', () => {
  const renderComponent = () =>
    render({
      components: { CBadge },
      template: `<CBadge>this is a badge</CBadge>`,
    })

  it('should render properly', async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('a11y test', async () => {
    await testA11y(renderComponent())
  })
})

describe('<CStack />', () => {
  const renderComponent = () =>
    render({
      components: { CHStack, CVStack, CStack },
      template: `
      <CStack><div>1</div><div>2</div></CStack>
      <CHStack><div>1</div><div>2</div></CHStack>
      <CVStack><div>1</div><div>2</div></CVStack>
      `,
    })

  it('should render properly', async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('a11y test', async () => {
    await testA11y(renderComponent())
  })
})

describe('<CAspectRatio />', () => {
  const renderComponent = () =>
    render({
      components: { CAspectRatio },
      template: `
      <c-aspect-ratio maxW="360px" :ratio="16 / 9" mb="4">
        <img src="https://bit.ly/naruto-sage" alt="img" />
      </c-aspect-ratio>
      `,
    })

  it('should render properly', async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('a11y test', async () => {
    await testA11y(renderComponent())
  })
})
