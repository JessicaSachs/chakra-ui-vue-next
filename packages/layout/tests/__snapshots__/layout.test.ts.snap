// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CBadge /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-badge css-1f1oimr"
  >
    this is a badge
  </div>
</DocumentFragment>
`;

exports[`<CLink /> should render properly 1`] = `
<DocumentFragment>
  <a
    class="chakra-link css-xisxf"
  >
    CLink
  </a>
  <a
    class="chakra-link css-xisxf"
    rel="noopener noreferrer"
    target="_blank"
  >
    CLink external
  </a>
</DocumentFragment>
`;

exports[`<CStack /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-x9juev"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
  <div
    class="chakra-stack-horizontal css-84zodg"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
  <div
    class="chakra-stack-vertical css-owjkmg"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</DocumentFragment>
`;
